From 6e759ba387f142a59edb1b3f70f1a2b3c710b3ab Mon Sep 17 00:00:00 2001
From: Jerome Duval <jerome.duval@gmail.com>
Date: Sun, 29 Nov 2015 15:49:57 +0000
Subject: Haiku patch

* Actually use pc_path when building pkgconfig.
* skip network libs in glib.

diff --git a/Makefile.am b/Makefile.am
index e46c798..b431bcb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -22,7 +22,7 @@ AM_CPPFLAGS = \
 	-DPKG_CONFIG_SYSTEM_LIBRARY_PATH="\"$(system_library_path)\""
 endif
 
-AM_CFLAGS = \
+AM_CFLAGS = $(AM_CPPFLAGS)\
 	$(WARN_CFLAGS) \
 	$(GCOV_CFLAGS) \
 	$(GLIB_CFLAGS)
diff --git a/glib/configure.ac b/glib/configure.ac
index ffa7869..48f7537 100644
--- a/glib/configure.ac
+++ b/glib/configure.ac
@@ -1094,49 +1094,6 @@ AS_IF([test $glib_native_win32 = yes], [
                                          [AC_MSG_ERROR([could not compile test program either way])])])])
   AC_SUBST(NAMESER_COMPAT_INCLUDE)
 
-  # We can't just use AC_CHECK_FUNC/AC_CHECK_LIB here. Bug 586150
-  NETWORK_LIBS=""
-  AC_MSG_CHECKING([for res_query])
-  AC_TRY_LINK([#include <sys/types.h>
-	       #include <netinet/in.h>
-	       #include <arpa/nameser.h>
-	       #include <resolv.h>
-              ],[
-	       res_query("test", 0, 0, (void *)0, 0);
-              ],[AC_MSG_RESULT([yes])],
-              [save_libs="$LIBS"
-               LIBS="-lresolv $LIBS"
-               AC_TRY_LINK([#include <sys/types.h>
-			    #include <netinet/in.h>
-			    #include <arpa/nameser.h>
-			    #include <resolv.h>
-			   ],[
-			    res_query("test", 0, 0, (void *)0, 0);
-			   ],[AC_MSG_RESULT([in -lresolv])
-                              NETWORK_LIBS="-lresolv $NETWORK_LIBS"],
-                           [LIBS="-lbind $save_libs"
-                            AC_TRY_LINK([#include <resolv.h>],
-                                        [res_query("test", 0, 0, (void *)0, 0);],
-                                        [AC_MSG_RESULT([in -lbind])
-                                         NETWORK_LIBS="-lbind $NETWORK_LIBS"],
-                                        [AC_MSG_ERROR(not found)])])
-               LIBS="$save_libs"])
-  AC_CHECK_FUNC(socket, :, AC_CHECK_LIB(socket, socket,
-				        [NETWORK_LIBS="-lsocket $NETWORK_LIBS"],
-				        [AC_MSG_ERROR(Could not find socket())]))
-  save_libs="$LIBS"
-  LIBS="$LIBS $NETWORK_LIBS"
-  AC_MSG_CHECKING([for res_init])
-  AC_TRY_LINK([#include <sys/types.h>
-	       #include <netinet/in.h>
-	       #include <arpa/nameser.h>
-	       #include <resolv.h>
-              ],[
-	       res_init();
-              ],[AC_MSG_RESULT([yes])
-	         AC_DEFINE(HAVE_RES_INIT, 1, [Define to 1 if you have the 'res_init' function.])
-	      ],[AC_MSG_RESULT([no])])
-  LIBS="$save_libs"
 ])
 AC_SUBST(NETWORK_LIBS)
 
-- 
2.2.2

